src/AutoRest.CSharp/Mgmt/Generation/ResourceWriter.cs

```diff
@@ -32,6 +32,11 @@ public void WriteResource(CodeWriter writer, Resource resource, BuildContext<Mgm
                     // inherits the default constructor when it is not a resource
                     var resourceData = context.Library.GetResourceData(resource.OperationGroup);
                     var baseConstructor = resourceData.IsResource() ? $" : base(options, resource.Id)" : string.Empty;
+                    if (!string.IsNullOrEmpty(baseConstructor) && resource.OperationGroup.IsSingletonResource(context.Configuration.MgmtConfiguration))
```

Comment:
does inheritance make more sense here vs the if statements?  I can see it going either way, but want to make sure we considered both options before landing on the if statements.

Reply:
For this simple one, I would say if statement is fine. For more **complex structure** differences I think reuse via inheritance would make more sense. Otherwise we will have to juggle the nuances of overriding code inside a functional emission block with passing in custom params.